src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/SNSService.kt

```diff
@@ -65,3 +68,31 @@ class SNSReferralService(
     }
   }
 }
+
+@Service
+class SNSAppointmentService(
+  private val snsPublisher: SNSPublisher,
+) : ApplicationListener<AppointmentEvent>, SNSService {
+
+  @AsyncEventExceptionHandling
+  override fun onApplicationEvent(event: AppointmentEvent) {
+    when (event.type) {
+      AppointmentEventType.ATTENDANCE_RECORDED -> {
+        val referral = event.appointment.actionPlan.referral
+        val eventType = "intervention.session-appointment.${when (event.appointment.attended) {
+          Attended.YES, Attended.LATE -> "attended"
+          Attended.NO -> "missed"
+          null -> throw RuntimeException("event triggered for appointment with no recorded attendance")
```

Comment:
When we get this exception, would we be able to see which appointment it relates to? Should we add a context (ID?) in the message?

🙋‍♂️ `IllegalStateException` may also fit the intent of the exception

Reply:
yes, all exceptions are handled by this class, which logs the event 

```@Aspect
@Component
class AsyncEventExceptionAdvice {
  companion object : KLogging()

  @Around(value = "@annotation(AsyncEventExceptionHandling)")
  fun handleException(pjp: ProceedingJoinPoint) {
    try {
      pjp.proceed()
    } catch (exception: Throwable) {
      val event = (pjp as MethodInvocationProceedingJoinPoint).args.getOrNull(0)
      logger.error("Exception thrown for method annotated with @AsyncEventExceptionHandling {}", exception, kv("event", event))
    }
  }
}```

Reply:
cool, thanks

Reply:
_not for this PR_
🙋‍♂️ `Throwable` might be dangerous to handle though... `Error` subclasses `Throwable` and it's generally advised to not catch those.